src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -561,9 +561,9 @@ trait Namers extends MethodSynthesis {
           }
         }
         if (!tree.symbol.isSynthetic && expr.symbol != null && !expr.symbol.isInterpreterWrapper) {
-          if (base.member(from) != NoSymbol)
+          if (base.member(from).exists)
             check(to0)
-          if (base.member(from.toTypeName) != NoSymbol)
+          if (base.member(from.toTypeName).exists)
```

Comment:
That's interesting, I always wonder about when it's OK to check exists. I have doubts about this "permanently hidden" check, which in the past was incorrect (for some case). The comment says "warn proactively", but there's no reason to be eager (though maybe there was in the past). An unused import is just an unused import; maybe an unused import of an implicit element should warn even if the lint is not enabled. Also ensure the warning is issued despite errors (due to missing implicits).

Other than deleting this check, maybe it's not terrible to force more, for the sake of more correctness.